packages/client/src/schedule-client.ts

```diff
@@ -0,0 +1,910 @@
+import { isServerErrorResponse, ServiceError } from './errors';
+import {
+  Backfill,
+  CalendarSpec,
+  CalendarSpecDescription,
+  CompiledScheduleOptions,
+  CompiledUpdatedSchedule,
+  compileScheduleOptions,
+  compileUpdatedScheduleOptions,
+  dayOfWeekNameToNumber,
+  dayOfWeekNumberToName,
+  IntervalSpecDescription,
+  ListScheduleEntry,
+  ListScheduleOptions,
+  monthNameToNumber,
+  monthNumberToName,
+  Range,
+  Schedule,
+  ScheduleActionType,
+  ScheduleOptions,
+  ScheduleOverlapPolicy,
+  UpdatedSchedule,
+  UpdateScheduleOptions,
+} from './schedule-types';
+import { Replace } from '@temporalio/common/lib/type-helpers';
+import { temporal } from '@temporalio/proto';
+import {
+  msOptionalToTs,
+  msToTs,
+  optionalDateToTs,
+  optionalTsToDate,
+  optionalTsToMs,
+  tsToDate,
+} from '@temporalio/common/lib/time';
+
+type CreateScheduleRequest = temporal.api.workflowservice.v1.ICreateScheduleRequest;
+// type CreateScheduleResponse = temporal.api.workflowservice.v1.ICreateScheduleResponse;
+type DescribeScheduleRequest = temporal.api.workflowservice.v1.IDescribeScheduleRequest;
+type DescribeScheduleResponse = temporal.api.workflowservice.v1.IDescribeScheduleResponse;
+type DeleteScheduleRequest = temporal.api.workflowservice.v1.IDeleteScheduleRequest;
+type DeleteScheduleResponse = temporal.api.workflowservice.v1.IDeleteScheduleResponse;
+type UpdateScheduleRequest = temporal.api.workflowservice.v1.IUpdateScheduleRequest;
+type UpdateScheduleResponse = temporal.api.workflowservice.v1.IUpdateScheduleResponse;
+type PatchScheduleRequest = temporal.api.workflowservice.v1.IPatchScheduleRequest;
+type PatchScheduleResponse = temporal.api.workflowservice.v1.IPatchScheduleResponse;
+type ListSchedulesRequest = temporal.api.workflowservice.v1.IListSchedulesRequest;
+type ListSchedulesResponse = temporal.api.workflowservice.v1.IListSchedulesResponse;
+
+// FIXME: Determine if we want to support these
+type ListScheduleMatchingTimesRequest = temporal.api.workflowservice.v1.IListScheduleMatchingTimesRequest;
+type ListScheduleMatchingTimesResponse = temporal.api.workflowservice.v1.IListScheduleMatchingTimesResponse;
+
+/**
+ * Handle to a single Schedule
+ */
+export interface ScheduleHandle {
+  /**
+   * This Schedule's identifier
+   */
+  readonly scheduleId: string;
+
+  /**
+   * Fetch the Schedule's description from the Server
+   */
+  describe(): Promise<Schedule>;
+
+  /**
+   * Update the Schedule
+   *
+   * Apply a user update function to the current schedule definition, and then try to update the
+   * schedule definition on the server. Note that, in the future, the user provided function might
+   * be invoked multiple time, with identical or different input.
+   *
+   * WARNING: At this moment, the server provides no garantee that update was applied sucessfully.
+   */
+  update<Action extends Workflow = Workflow>(updateFn: (previous: Schedule) => UpdatedSchedule<Action>): Promise<void>;
```

Comment:
```suggestion
  /**
   * Update the Schedule
   *
   * This function calls `.describe()`, provides the `Schedule` to the provided `updateFn`, and sends the returned `UpdatedSchedule` to the Server to update the Schedule definition. Note that, in the future, `updateFn` might
   * be invoked multiple times, with identical or different input.
   *
   * WARNING: At this moment, the server provides no guarantee that an update is applied successfully.
   */
  update<Action extends Workflow = Workflow>(updateFn: (previous: Schedule) => UpdatedSchedule<Action>): Promise<void>;
```

Are we including `conflict_token` in `UpdateScheduleRequest`? If not, I'd remove the WARNING, since the update will always be applied.